arangod/Replication2/ReplicatedLog/Supervision.cpp

```diff
@@ -382,38 +382,44 @@ auto getRemovedParticipant(ParticipantsFlagsMap const &targetParticipants,
 //
 // These actions are executes by using std::visit via an Executor struct that
 // contains the necessary context.
-auto checkReplicatedLog(LogTarget const &target,
-                        std::optional<LogPlanSpecification> const &maybePlan,
-                        std::optional<LogCurrent> const &maybeCurrent,
-                        ParticipantsHealth const &health) -> Action {
+auto checkReplicatedLog(LogTarget const& target,
+                        std::optional<LogPlanSpecification> const& maybePlan,
+                        std::optional<LogCurrent> const& maybeCurrent,
+                        ParticipantsHealth const& health) -> Action {
   if (!maybePlan) {
     // The log is not planned right now, so we create it
-    LogPlanTermSpecification term;
-    term.config = target.config;
-    term.term = LogTerm{1};
-    auto leader = target.participants.begin()->first;
-    auto rebootId = health._health.at(leader).rebootId;
-    term.leader.emplace(std::move(leader), rebootId);
-    return AddLogToPlanAction(target.id, target.participants, term);
+    // provided we have enough participants
+    if (target.participants.size() + target.config.writeConcern <
+        target.config.replicationFactor + 1) {
+      return ErrorAction(
+          LogCurrentSupervisionError::TARGET_NOT_ENOUGH_PARTICIPANTS);
+    } else {
+      auto leader = target.participants.begin()->first;
+      auto rebootId = health._health.at(leader).rebootId;
```

Comment:
I know it was me who came up with that. But maybe we want to select the leader randomly? What about the case that someone specified a leader explicitly in Target? Right now, we would select the first and then do a controlled failover. 

Would that be hard to add?